chia/full_node/full_node.py

```diff
@@ -1606,44 +1606,45 @@ async def respond_transaction(
         if self.sync_store.get_sync_mode():
             status = MempoolInclusionStatus.FAILED
             error: Optional[Err] = Err.NO_TRANSACTIONS_WHILE_SYNCING
+            self.mempool_manager.remove_seen(spend_name)
```

Comment:
so, not all changes to the mempool require the lock then? why is this safe?

Reply:
The point of the "seen" list is to prevent processing a transaction twice. We might get the same transaction many times from different peers, so this prevents repeating work. Also, it is safe because it does not use `await`, so it's blocking.

Reply:
It's just a cache that says: we received this transaction so no point in retrying it. If we add the transaction and then there is a reorg, we will remove it from the seen list so it can be processed again.

Reply:
if this is meant to undo the effect of:

```
self.mempool_manager.add_and_maybe_pop_seen(spend_name)
```

and there's no `await` call made in between (i.e. nothing can change "under our feed"). Why not just check `get_sync_mode()` up front, and never call `self.mempool_manager.add_and_maybe_pop_seen()` to begin with, in sync mode.

Reply:
actually, we already check this on line 1192. Is it possible for us to get into this code path?
Perhaps this should be an `assert` instead.